Jellyfin.Api/Controllers/LyricController.cs

```diff
@@ -0,0 +1,252 @@
+        IProviderManager providerManager,
+        IFileSystem fileSystem,
+        IUserManager userManager)
+    {
+        _libraryManager = libraryManager;
+        _lyricManager = lyricManager;
+        _providerManager = providerManager;
+        _fileSystem = fileSystem;
+        _userManager = userManager;
+    }
+
+    /// <summary>
+    /// Gets an item's lyrics.
+    /// </summary>
+    /// <param name="itemId">Item id.</param>
+    /// <response code="200">Lyrics returned.</response>
+    /// <response code="404">Something went wrong. No Lyrics will be returned.</response>
+    /// <returns>An <see cref="OkResult"/> containing the item's lyrics.</returns>
+    [HttpGet("Audio/{itemId}/Lyrics")]
+    [Authorize]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public async Task<ActionResult<LyricModel>> GetLyrics([FromRoute, Required] Guid itemId)
+    {
+        var isApiKey = User.GetIsApiKey();
+        var userId = User.GetUserId();
+        if (!isApiKey && userId.IsEmpty())
+        {
+            return BadRequest();
+        }
+
+        var audio = _libraryManager.GetItemById<Audio>(itemId);
+        if (audio is null)
+        {
+            return NotFound();
+        }
+
+        if (!isApiKey)
+        {
+            var user = _userManager.GetUserById(userId);
+            if (user is null)
+            {
+                return NotFound();
+            }
+
+            // Check the item is visible for the user
+            if (!audio.IsVisible(user))
+            {
+                return Unauthorized($"{user.Username} is not permitted to access item {audio.Name}.");
+            }
+        }
+
+        var result = await _lyricManager.GetLyricsAsync(audio, CancellationToken.None).ConfigureAwait(false);
+        if (result is not null)
+        {
+            return Ok(result);
+        }
+
+        return NotFound();
+    }
+
+    /// <summary>
+    /// Upload an external lyric file.
+    /// </summary>
+    /// <param name="itemId">The item the lyric belongs to.</param>
+    /// <param name="body">The request body.</param>
+    /// <response code="204">Lyrics uploaded.</response>
+    /// <response code="400">Error processing upload.</response>
+    /// <response code="404">Item not found.</response>
+    /// <returns>The uploaded lyric.</returns>
+    [HttpPost("Audio/{itemId}/Lyrics")]
+    [Authorize(Policy = Policies.LyricManagement)]
+    [ProducesResponseType(StatusCodes.Status204NoContent)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<ActionResult<LyricModel>> UploadLyric(
+        [FromRoute, Required] Guid itemId,
+        [FromBody, Required] UploadLyricDto body)
+    {
+        var audio = _libraryManager.GetItemById<Audio>(itemId);
+        if (audio is null)
+        {
+            return NotFound();
+        }
+
+        var stream = new MemoryStream(Encoding.UTF8.GetBytes(body.Data));
```

Comment:
```suggestion
        var bytes = Encoding.UTF8.GetBytes(body.Data);
        var stream = new MemoryStream(bytes, 0, bytes.Length, false, true);
```
We can expose the internal buffer for this MemoryStream, may speed something up down the line